### PR TITLE
Improve menu nav and pause ux

### PR DESCRIPTION
**Pull request by nathanleiby**
A handful of bugfixes and UX improvements:

- https://trello.com/c/YygyhVQC/43-cleverfall-unable-to-interact-with-pause-menu-during-combat
- https://trello.com/c/E2ThJQPc/53-cleverfall-bug-pause-game-before-entering-round-end-screen
- https://trello.com/c/fHfwF3Tj/54-allow-exiting-end-of-round-screen-via-next-button

Big changes are:
- mouse no longer can be used, to avoid accidental menu (un)select issues... we expect to play with gamepad, though keyboard is still supported.
- PauseMenu and EndCombatUI get instantiated on demand, rather than existing and show/hide via `SetActive(bool)`. This works better with the EventSystem, which is now able to select a default button reliably.

See commits for other changes.

### CODE REVIEW

**Comment by nathanleiby** [comment]
Beyond UX improvements, this should make it possible for us to playtest better today. It's now possible to play multiple combat rounds without quitting and restarting the executable :D

**Comment by nathanleiby** [comment]
menunav-improvements

**Comment by nathanleiby** [comment]
Gonna merge so this goes into our gameday build. Happy to respond to feedback post hoc

File: Assets/Scripts/CombatManager.cs [hunk 1]
```diff
@@ -66,22 +71,64 @@ public class CombatManager : MonoBehaviour {
...
     void endCombat() {
-        Debug.Log("CombatManager:EndCombat()");
+        Debug.Log("CombatManager:endCombat()");
+        combatIsOver = true;
+
+        Pause(true); // stop combat from running and further game interaction
 
         combatGUI.Hide();
+
+        endCombatUI = Instantiate(endCombatUIPrefab);
```
Comment (peternga) [inline Assets/Scripts/CombatManager.cs:81]: Does this ever get cleaned up?

Comment (nathanleiby) [reply Assets/Scripts/CombatManager.cs:81]: I believe when scene is unloaded (moving from level => menu) this gets Destroyed, but we could be more explicit

**Pull request merged**